elfpy/utils/apeworx_integrations.py

```diff
@@ -0,0 +1,250 @@
+        prefix, maturity_timestamp = decode_asset_id(token_id)
+        pool_state = hyperdrive_contract.getPoolInfo().__dict__  # type: ignore
+        pool_state["block_number_"] = tx_receipt.block_number
+        pool_state["token_id_"] = token_id
+        pool_state["prefix_"] = prefix
+        pool_state["maturity_timestamp_"] = maturity_timestamp
+        logging.debug("hyperdrive_pool_state=%s", pool_state)
+    return pool_state, tx_receipt
+
+
+def ape_close_position(
+    trade_prefix: AssetIdPrefix,
+    hyperdrive_contract: ContractInstance,
+    agent_address: AccountAPI,
+    bond_amount: int,
+    maturity_time: int,
+) -> tuple[dict[str, Any], Receipt]:
+    r"""Close a long or short position on the Hyperdrive Solidity contract using apeworx.
+
+    Arguments
+    ---------
+    trade_prefix: AssetIdPrefix
+        IntEnum specifying whether the trade is a long (0) or a short (1).
+    hyperdrive_contract: ape.contracts.base.ContractInstance
+        Ape project `ContractInstance
+        <https://docs.apeworx.io/ape/stable/methoddocs/contracts.html#ape.contracts.base.ContractInstance>`_
+        wrapped around the initialized MockHyperdriveTestnet smart contract.
+    agent_address: ape.api.accounts.AccountAPI
+        Ape address container, or `AccountAPI
+        <https://docs.apeworx.io/ape/stable/methoddocs/api.html#ape.api.accounts.AccountAPI>`_
+        representing the agent which is executing the action.
+    bond_amount: int
+        Unsigned int-256 representation of the bond amount that the agent wishes to sell.
+    maturity_time: int
+        Unsigned int-256 representation of the maturity time of the short being sold.
+        This is equal to the pool position duration plus the checkpoint time that is
+        closest to (but before) the corresponding open trade.
+
+    Returns
+    -------
+    Tuple[dict[str, Any], ape_ethereum.transactions.Receipt]
+        A tuple containing an update dictionary for the Hyperdrive pool state
+        as well as the Ape-ethereum transaction `receipt
+        <https://docs.apeworx.io/ape/stable/methoddocs/api.html#ape.api.transactions.ReceiptAPI>`_.
+    """
+    # pylint: disable=too-many-locals
+    with ape.accounts.use_sender(agent_address):  # sender for contract calls
+        # Ensure requested close amount is not greater than what is available in the pool
+        trade_asset_id = encode_asset_id(trade_prefix, maturity_time)
+        agent_balance = hyperdrive_contract.balanceOf(trade_asset_id, agent_address)  # type: ignore
+        if bond_amount < agent_balance:
+            trade_bond_amount = bond_amount
+        else:
+            trade_bond_amount = agent_balance
+            logging.warning(
+                "bond_amount=%g is greater than or equal to the Hyperdrive pool balance=%g",
+                bond_amount,
+                trade_bond_amount,
+            )
+        # Close the position
+        min_output = 0
```

Comment:
ok so we allowing for maximum slippage here.  I would make this a function argument that has `min_output = 0` for now.